### ISSUE TITLE
Flux v2.2 support - podinfo-hr stuck in progressing

### ISSUE THREAD

**Issue by demisx** [flux-subsystem-argo/flamingo#51]
I've followed the instructions in readme. Everything went smoothly, but the `podinfo-hr` seems to be stuck in "progressing" state. 

<img width="788" alt="Screenshot 2024-01-31 at 9 58 59 AM" src="https://github.com/flux-subsystem-argo/flamingo/assets/4452412/b4d77384-8775-445c-99e7-2ae8aad1a6ce">

```yaml
$ kubectl describe applications -n argocd  podinfo-hr
---
Name:         podinfo-hr
Namespace:    argocd
Labels:       app.kubernetes.io/managed-by=flamingo
              flamingo/destination-namespace=podinfo-helm
              flamingo/source-type=HelmRepository
              flamingo/workload-name=podinfo
              flamingo/workload-type=HelmRelease
Annotations:  weave.gitops.flamingo/base-url: http://localhost:9001
              weave.gitops.flamingo/cluster-name: Default
API Version:  argoproj.io/v1alpha1
Kind:         Application
Metadata:
  Creation Timestamp:  2024-01-31T17:37:53Z
  Generation:          16
  Resource Version:    4646
  UID:                 d98adc9e-fa49-4159-870b-2c1891721479
Spec:
  Destination:
    Namespace:  podinfo-helm
    Server:     https://kubernetes.default.svc
  Project:      default
  Source:
    Chart:  podinfo
    Helm:
      Release Name:   podinfo
    Repo URL:         oci://ghcr.io/stefanprodan/charts
    Target Revision:  *
  Sync Policy:
    Sync Options:
      ApplyOutOfSyncOnly=true
      FluxSubsystem=true
Status:
  Controller Namespace:  argocd
  Health:
    Status:  Progressing
  History:
    Deploy Started At:  2024-01-31T17:43:12Z
    Deployed At:        2024-01-31T17:43:12Z
    Id:                 0
    Source:
      Chart:  podinfo
      Helm:
        Release Name:   podinfo
      Repo URL:         oci://ghcr.io/stefanprodan/charts
      Target Revision:  *
  Operation State:
    Finished At:  2024-01-31T17:43:12Z
    Message:      successfully synced (no more tasks)
    Operation:
      Initiated By:
        Username:  admin
      Retry:
      Sync:
        Revision:  *
        Sync Options:
          ApplyOutOfSyncOnly=true
          FluxSubsystem=true
        Sync Strategy:
          Hook:
    Phase:       Succeeded
    Started At:  2024-01-31T17:43:12Z
    Sync Result:
      Revision:  
      Source:
        Chart:  podinfo
        Helm:
          Release Name:   podinfo
        Repo URL:         oci://ghcr.io/stefanprodan/charts
        Target Revision:  *
  Reconciled At:          2024-01-31T18:01:30Z
  Resources:
    Group:  helm.toolkit.fluxcd.io
    Health:
      Message:  Helm install succeeded for release podinfo-helm/podinfo-helm-podinfo.v1 with chart podinfo@6.5.4
      Status:   Healthy
    Kind:       HelmRelease
    Name:       podinfo
    Namespace:  podinfo-helm
    Status:     Synced
    Version:    v2beta1
    Group:      source.toolkit.fluxcd.io
    Health:
      Message:  Waiting for HelmRepository to reconcile ...
      Status:   Progressing
    Kind:       HelmRepository
    Name:       podinfo
    Namespace:  podinfo-helm
    Status:     Synced
    Version:    v1beta2
  Summary:
  Sync:
    Compared To:
      Destination:
        Namespace:  podinfo-helm
        Server:     https://kubernetes.default.svc
      Source:
        Chart:  podinfo
        Helm:
          Release Name:   podinfo
        Repo URL:         oci://ghcr.io/stefanprodan/charts
        Target Revision:  *
    Status:               Synced
Events:
  Type    Reason              Age   From                           Message
  ----    ------              ----  ----                           -------
  Normal  ResourceUpdated     24m   argocd-application-controller  Updated sync status:  -> Synced
  Normal  ResourceUpdated     24m   argocd-application-controller  Updated health status:  -> Progressing
  Normal  OperationStarted    19m   argocd-server                  admin initiated sync to *
  Normal  OperationCompleted  19m   argocd-application-controller  Sync operation to  succeeded
```

**Comment by chanwit** [comment]
It's because of Flux v2.2.x and the Helm apiVersion being implicitly promoted to v2beta2, which we haven't supported yet.
We'll update Flamingo to support them shortly.

**Comment by chanwit** [comment]
Fixed in v0.8.2

![image](https://github.com/flux-subsystem-argo/flamingo/assets/10666/21879b77-e107-45d6-9552-930f447a7fea)
